### PR TITLE
Remove obsolete workarounds.

### LINKED ISSUE #239: Remove probably-obsolete fixes for (gribapi bugs in signed/unsiged keys
Review all usage (and tesing) of [_load_convert::fixup_float32_from_int32](https://github.com/SciTools/iris-grib/blob/master/iris_grib/_load_convert.py#L231)

Probably the original problems have all gone away by now ?

### PR DESCRIPTION
**Pull request by pp-mo**
Closes #239 

I'm happy that none of these workarounds are now needed, as all fixed in eccodes.  Long ago, I think.
This passes existing tests, but there is no effective testing of the relevant code.
What I have done is a roundtrip test  : that at least **_does_** seem to work :

```
>>> ellipsoid = GeogCS(EARTH_RADIUS)
>>> data = np.arange(12).reshape(3, 4)
>>> cube = Cube(data, "air_pressure_anomaly")
>>> trans_merc = TransverseMercator(
...     49.0, -2.0, -400000.0, 100000.0, 0.9996012717, ellipsoid
... )
>>> coord = DimCoord(
...     np.arange(3),
...     "projection_y_coordinate",
...     units="m",
...     coord_system=trans_merc,
... )
>>> cube.add_dim_coord(coord, 0)
>>> coord = DimCoord(
...     np.arange(4),
...     "projection_x_coordinate",
...     units="m",
...     coord_system=trans_merc,
... )
>>> cube.add_dim_coord(coord, 1)
>>> coord = DimCoord(
...     np.arange(1),
...     "time",
...     units="days since 2000-01-01",
... )
>>> cube.add_aux_coord(coord, ())
>>> 
>>> print("\nInital cube to save:"); print(cube)

Inital cube to save:
air_pressure_anomaly / (unknown)    (projection_y_coordinate: 3; projection_x_coordinate: 4)
    Dimension coordinates:
        projection_y_coordinate                             x                           -
        projection_x_coordinate                             -                           x
    Scalar coordinates:
        time                        2000-01-01 00:00:00
>>> print("\nsaved + reloaded ...\n")

saved + reloaded ...

>>> result = iris.load_cube('temp.grib2')
>>> print(result); print("  ==> re-loaded coord system = ", result.coord_system())
unknown / (unknown)                 (projection_y_coordinate: 3; projection_x_coordinate: 4)
    Dimension coordinates:
        projection_y_coordinate                             x                           -
        projection_x_coordinate                             -                           x
    Scalar coordinates:
        forecast_period             0 hours
        forecast_reference_time     2000-01-01 00:00:00
        time                        2000-01-01 00:00:00
    Attributes:
        GRIB_PARAM                  GRIB2:d255c255n255
  ==> re-loaded coord system =  TransverseMercator(latitude_of_projection_origin=49.0, longitude_of_central_meridian=-2.0, false_easting=-400000.0, false_northing=100000.0, scale_factor_at_central_meridian=0.9996012449264526, ellipsoid=GeogCS(6371229.0))
>>> 
```

### CODE REVIEW

**Review by stephenworsley — CHANGES_REQUESTED** [review]
`_save_rules` also defines `fixup_int32_as_int32` and `fixup_int32_as_uint32`, should these also be removed like`fixup_float32_from_int32`? Their docstring also mentions how they are a workaround, is this for the same issue?

File: iris_grib/_load_convert.py [hunk 1]
```diff
@@ -848,19 +814,17 @@ def grid_definition_template_12(section, metadata):
...
     # Catch bug in ECMWF GRIB API (present at 1.12.1) where the scale
     # is treated as a signed, 4-byte integer.
-    if isinstance(scale, int):
-        scale = fixup_float32_from_int32(scale)
     CM_TO_M = 0.01
     easting = section["XR"] * CM_TO_M
     northing = section["YR"] * CM_TO_M
     cs = icoord_systems.TransverseMercator(lat, lon, easting, northing, scale, geog_cs)
 
     # Deal with bug in ECMWF GRIB API (present at 1.12.1) where these
```
Comment (stephenworsley) [inline iris_grib/_load_convert.py:822]: This comment can be removed now

File: iris_grib/_save_rules.py [hunk 2]
```diff
@@ -608,9 +549,6 @@ def m_to_cm(value):
     # but it should accept a float, so work around this.
```
Comment (stephenworsley) [inline iris_grib/_save_rules.py:549]: I believe this comment is no longer applicable.

File: iris_grib/_save_rules.py [hunk 3]
```diff
@@ -578,10 +519,10 @@ def grid_definition_template_12(cube, grib):
     # GRIBAPI expects unsigned ints in X1, X2, Y1, Y2 but it should accept
```
Comment (stephenworsley) [inline iris_grib/_save_rules.py:519]: This comment can be removed

**Review by stephenworsley — CHANGES_REQUESTED** [review]
Looks good, there's a couple comments which ought to be removed and this should be good to go.

**Comment by pp-mo** [comment]
Thanks @stephenworsley good spots !
I think I addressed those 3 now, please take another look.

**Review by stephenworsley — APPROVED** [review]
Looks good.

**Comment by pp-mo** [comment]
Thanks @stephenworsley  !

**Pull request merged**